differ/differ.go

```diff
@@ -301,6 +309,97 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
 	return evaluator.Evaluate(eventFilter, values)
 }
 
+func findRenderedReport(reports []types.RenderedReport, ruleName types.RuleName, errorKey types.ErrorKey) (types.RenderedReport, error) {
+	for _, report := range reports {
+		reportRuleName := string(report.RuleID)[strings.LastIndex(string(report.RuleID), ".")+1:]
+		if reportRuleName == string(ruleName) && report.ErrorKey == errorKey {
+			return report, nil
+		}
+	}
+	return types.RenderedReport{}, errors.New("report for given rule ID nad error key has not been found")
+}
+
+func createServiceLogEntry(report types.RenderedReport, cluster types.ClusterEntry) types.ServiceLogEntry {
+	logEntry := types.ServiceLogEntry{
+		ClusterUUID: string(cluster.ClusterName),
+		Description: report.Description,
+		ServiceName: "CCX Notification Service",
+		Summary:     report.Reason,
+	}
+
+	// It is necessary to truncate the fields because of Service Log limitations
+	if len(logEntry.Description) > serviceLogMaxTextLength {
+		logEntry.Description = logEntry.Description[:serviceLogMaxTextLength]
+	}
+
+	if len(logEntry.Summary) > serviceLogMaxTextLength {
+		logEntry.Summary = logEntry.Summary[:serviceLogMaxTextLength]
+	}
+
+	return logEntry
+}
+
+func produceEntriesToServiceLog(config conf.ConfigStruct, cluster types.ClusterEntry,
```

Comment:
this looks like the `produceEntriesToKafka` function. Do you have plans on merging both? what will happen if a user run the service with both service log and kafka enabled? we will duplicate many work, right?

Reply:
There is some common code (finding rule by name and error key, assembling event value), on the other hand, message is assembled differently for Kafka and Service Log, and the message is sent at different place in code. Should I place these into the same loop?

Reply:
> finding rule by name and error key, assembling event value

this part is not fully equal, right? there is a tiny difference in the event_target column

> Should I place these into the same loop?

not sure TBH.Will using the same improve performance? would we skip some operations?

Reply:
I believe the performance will not be worse by much (the common code should not computationally expensive) and I cannot think of easy and readible way of how to refactor this code at the moment.

Reply:
dont worry :slightly_smiling_face: